scheduling/task_run.go

```diff
@@ -0,0 +1,160 @@
+	dummyTrigger  TriggerType = "DummyTrigger"
+)
+
+type Trigger interface {
+	Type() TriggerType
+	Name() string
+}
+
+type OneOffTrigger struct {
+	TriggerName string      `json:"triggerName"`
+	TriggerType TriggerType `json:"triggerType"`
+}
+
+func (t OneOffTrigger) Type() TriggerType {
+	return t.TriggerType
+}
+
+func (t OneOffTrigger) Name() string {
+	return t.TriggerName
+}
+
+type DummyTrigger struct {
+	TriggerName string      `json:"triggerName"`
+	TriggerType TriggerType `json:"triggerType"`
+	DummyField  bool        `json:"dummyField"`
+}
+
+func (t DummyTrigger) Type() TriggerType {
+	return t.TriggerType
+}
+
+func (t DummyTrigger) Name() string {
+	return t.TriggerName
+}
+
+type TaskRunMetadata struct {
+	ID        TaskRunId `json:"id"`
+	RunId     TaskId    `json:"runId"`
+	Name      string    `json:"name"`
+	Trigger   Trigger   `json:"trigger"`
+	Status    Status    `json:"status"`
+	StartTime time.Time `json:"startTime"`
+	EndTime   time.Time `json:"endTime"`
+	Logs      []string  `json:"logs"`
+	Error     string    `json:"error"`
+}
+
+// Read Methods
+// func (t *TaskRunMetadata) ID() TaskRunId
+// func (t *TaskRunMetadata) RunID() TaskId
+// func (t *TaskRunMetadata) Name() string
+// func (t *TaskRunMetadata) Trigger() Trigger
+// func (t *TaskRunMetadata) Status() Status // Enum
+// func (t *TaskRunMetadata) StartTime() Time
+// func (t *TaskRunMetadata) EndTime() Time
+// func (t *TaskRunMetadata) Logs() []string
+// func (t *TaskRunMetadata) Error() string
+
+// Formatting
+func (t *TaskRunMetadata) ToJSON() ([]byte, error) {
+	return json.Marshal(t)
+}
+func (t *TaskRunMetadata) FromJSON(data []byte) error {
+	type tempConfig struct {
+		ID        TaskRunId       `json:"id"`
+		RunId     TaskId          `json:"runId"`
+		Name      string          `json:"name"`
+		Trigger   json.RawMessage `json:"trigger"`
+		Status    Status          `json:"status"`
+		StartTime time.Time       `json:"startTime"`
+		EndTime   time.Time       `json:"endTime"`
+		Logs      []string        `json:"logs"`
+		Error     string          `json:"error"`
+	}
+
+	var temp tempConfig
+	if err := json.Unmarshal(data, &temp); err != nil {
+		return fmt.Errorf("failed to deserialize task run metadata: %w", err)
+	}
+
+	if temp.ID == 0 {
+		return fmt.Errorf("task run metadata is missing ID")
+	}
+	t.ID = temp.ID
+
+	if temp.RunId == 0 {
+		return fmt.Errorf("task run metadata is missing RunID")
+	}
+	t.RunId = temp.RunId
+
+	if temp.Name == "" {
+		return fmt.Errorf("task run metadata is missing required fields")
+	}
+	t.Name = temp.Name
+
+	if temp.Status == "" || (temp.Status != Success && temp.Status != Failed && temp.Status != Pending) {
+		return fmt.Errorf("unknown status: %s", temp.Status)
+	}
+	t.Status = temp.Status
+
+	if temp.StartTime.IsZero() {
+		return fmt.Errorf("task run metadata is missing Start Time")
+	}
+	t.StartTime = temp.StartTime
+
+	if temp.EndTime.IsZero() {
+		return fmt.Errorf("task run metadata is missing End Time")
```

Comment:
for errors, we might have to leverage the errors package if it gets merged.

Reply:
Yeah, im leaving the errors as-is for now, but once the errors package gets merged we need to swap these out